tests/test-bulk.c

```diff
@@ -1557,7 +1557,7 @@ _test_write_concern (bool has_write_commands, bool ordered)
       request = mock_server_receives_command (
          mock_server,
          "test",
-         MONGOC_QUERY_NONE,
+         MONGOC_QUERY_SLAVE_OK,
```

Comment:
It seems minorly buggy to me that the driver now sets SlaveOk on write commands. Server Selection Spec: ["clients MUST always set the slaveOK wire protocol flag on reads"](https://github.com/mongodb/specifications/blob/master/source/server-selection/server-selection.rst#topology-type-single). It's reasonable to infer that it shouldn't set SlaveOk on writes, and other drivers (PyMongo) don't, that I know of.

Worse, I think the driver will now send $readPreference in write commands when connected to a single mongos?

The reason it didn't set SlaveOk / $readPreference on writes before, is because write commands set the cursor's hint:

https://github.com/mongodb/mongo-c-driver/blob/1941a0a6fd7271ac26d7a3a152df30f810a662a8/src/mongoc/mongoc-client.c#L1186-L1186

The hint comes from a call to _mongoc_client_preselect up in _mongoc_write_command_execute. Call stack, most recent call on top:
```
_mongoc_cursor_query mongoc-cursor.c:602
_mongoc_cursor_next mongoc-cursor.c:948
mongoc_cursor_next mongoc-cursor.c:876
_mongoc_client_command_simple_with_hint mongoc-client.c:1188
_mongoc_write_command mongoc-write-command.c:889
_mongoc_write_command_execute mongoc-write-command.c:961
```
It works by accident: since write commands always set a hint on the cursor, they aren't sent with SlaveOk.

Now, I think, we need a private bool member is_write_command, and a private function 
```
_mongoc_cursor_set_write_command (cursor, bool)
```
If the bool is set, don't apply read preferences. 

I guess _mongoc_client_command_simple_with_hint needs an is_write_command parameter now, too, so _mongoc_write_command can pass it as "true".